docs/src/faqs/migrate_from_fnapi.rst

```diff
@@ -1,229 +1,76 @@
-    Do not remove the ``worker_harness_container_image`` value under ``pipeline_options``. Klio
-    uses the image as a `"driver"`_ for Beam.
-
 
 Update: ``Dockerfile``
 ^^^^^^^^^^^^^^^^^^^^^^
 
 Required Changes
 ~~~~~~~~~~~~~~~~
 
-1. **ADD** ``klio-job.yaml`` to be copied into ``/usr/src/app``.
-
-    .. collapsible:: Why is this needed?
 
-        We need to include Klio's configuration, but when creating a package of the job, the configuration must be within the same directory ``setup.py`` is in (subdirectories are fine). Relatedly, multi-configuration is not yet supported without the FnAPI since Klio expects the job configuration in a location that we can't manipulate with the ``setup.py`` approach.
-
-2. **ADD** the newly required files to be copied over - ``setup.py`` and ``MANIFEST.in`` - into the working directory, ``/usr/src/app``.
-
-    .. collapsible:: Why is this needed?
-
-        ``setup.py`` and ``MANIFEST.in`` are needed to tell Klio and Dataflow how to build your pipeline as a Python package (i.e. what Python and non-Python files to include) since you're no longer using a Docker image as a "package" for your job.
-
-3. **DOUBLE CHECK** any non-Python files needed for the job, e.g. models, JSON schemas, etc, are copied into the working directory, ``/usr/src/app``.
-
-    .. collapsible:: Why is this needed?
-
-        Klio packages up your job to be installed (for unit tests, audits, and running on the direct runner), and to be uploaded to Dataflow locally on the job's worker image. Therefore, the Docker image needs to have all the required Python and non-Python files to run the job.
-
-4. **ADD** the following line to the end of the file: ``RUN pip install .``
+1. **UPDATE**  ``RUN pip install .`` to ``RUN pip install -r job-requirements.txt``
 
     .. collapsible:: Why is this needed?
 
-        We install the package for the ability to run unit tests via ``klio job test``, run audits via ``klio job audit``, and - if needed - to run the job with Direct Runner.
+       We now install dependencies directly on the worker image.
 
-5. **DOUBLE CHECK** that you ``COPY`` in your ``job-requirements.txt`` file into the image (it should already exist if the job was made via ``klio job create``). It can be grouped into one ``COPY`` line like the example below.
+2. **DOUBLE CHECK** that you ``COPY`` in your ``job-requirements.txt`` file into the image (it should already exist if the job was made via ``klio job create``). It can be grouped into one ``COPY`` line like the example below.
 
 .. collapsible:: Example of Required Changes
 
     .. code-block:: diff
 
           COPY __init__.py \
-        +     setup.py \
-        +     MANIFEST.in \
         +     my-model.h5 \
-        +     klio-job.yaml \
         +     job-requirements.txt \
               run.py \
               transforms.py \
               /usr/src/app/
 
-        + RUN pip install .
+        - RUN pip install .
+        + RUN pip install -r job-requirements.txt
```

Comment:
Originally, `RUN pip install .` is at the end to install the package (as defined in `setup.py`). But when using the FnAPI, it should be before the copying, since the files in `COPY` are more likely to change than `job-requirements.txt` (and therefore it's more performant to pip install deps earlier). So it actually should be something like:

```Dockerfile
...
COPY job-requirements.txt job-requirements.txt
RUN pip install -r job-requirements.txt

COPY __init__.py \
    my-model.h5 \
    run.py \
    transforms.py \
    /usr/src/app/
```